File: app/src/main/java/com/gigigo/orchextra/ocm/sample/ocm/OcmWrapper.java
package com.gigigo.orchextra.ocm.sample.ocm;

public interface OcmWrapper {

  boolean isOcmInitialized();

  void startWithCredentials(String apiKey, String apiSecret, String businessUnit,
      String vimeoAccessToken, OnStartWithCredentialsCallback onStartWithCredentialsCallback);

  void setUserIsAuthorizated(boolean isUserLogged);

  void setContentLanguage(String languageCode);

  void scanCode(ScanCodeListener scanCodeListener);

  void processDeepLink(String deepLink);

  interface OnStartWithCredentialsCallback {
    void onCredentialReceiver(String accessToken);

    void onCredentailError();
  }

  interface StatusListener {
    void onSuccess();

    void onError(String error);
  }

  interface ScanCodeListener {
    void onCodeScan(String code);
  }
}

File: showcase/src/main/java/com/gigigo/showcase/ocm/OcmWrapper.kt
package com.gigigo.showcase.ocm


interface OcmWrapper {

  val isOcmInitialized: Boolean

  fun startWithCredentials(apiKey: String, apiSecret: String, businessUnit: String,
      onStartWithCredentialsCallback: OnStartWithCredentialsCallback)

  fun setUserIsAuthorizated(isUserLogged: Boolean)

  fun setContentLanguage(languageCode: String)

  fun scanCode(scanCodeListener: ScanCodeListener)

  interface OnStartWithCredentialsCallback {
    fun onCredentialReceiver(accessToken: String)

    fun onCredentailError()
  }

  interface StatusListener {
    fun onSuccess()

    fun onError(error: String)
  }

  interface ScanCodeListener {
    fun onCodeScan(code: String)
  }
}
